boards/m5stamp-c3.json

```diff
@@ -0,0 +1,33 @@
+{
+  "build": {
+    "arduino":{
+      "ldscript": "esp32c3_out.ld"
+    },
+    "core": "esp32",
+    "f_cpu": "160000000L",
+    "f_flash": "40000000L",
+    "flash_mode": "dio",
```

Comment:
Please use mode `qio` Board does support

Reply:
Yes, `dio` was needed for core < 2.0.4. Was a bug in bootloader generation.

Reply:
I have pushed that, although I wasn't able to validate it yet because I'm using the commit based on the feature/arduino-upstream branch, which doesn't seem to have been worked on recently? Arduino support is quite important to me..